src/System.IO.Compression.Brotli/System/IO/Compression/Brotli.cs

```diff
@@ -15,8 +15,95 @@ namespace System.IO.Compression
 {
     public static class Brotli
     {
-        const int DefaultQuality = 11;
-        const int DefaultWindowSize = 24;
+        private const int MinWindowBits = 10;
+        private const int MaxWindowBits = 24;
+        private const int MinQuality = 0;
+        private const int MaxQuality = 11;
+
+        public struct State : IDisposable
+        {
+            internal IntPtr BrotliNativeState { get; private set; }
+            internal BrotliDecoderResult LastDecoderResult;
+            public bool CompressMode { get; set; }
```

Comment:
This should probably be readonly and set internally when InitializeDecoder or InitializeEncoder is called.